tools/pipelines/templates/build-npm-package.yml

```diff
@@ -407,6 +412,41 @@ extends:
                         inputs:
                           sourceFolder: '${{ parameters.buildDirectory }}/nyc/report'
                           targetFolder: $(Build.ArtifactStagingDirectory)/codeCoverageAnalysis
+                      - task: Bash@3
+                        displayName: Report Code Coverage Comparison
+                        condition: and(succeededOrFailed(), eq('${{ parameters.reportCodeCoverageComparison }}', true), eq(variables['ReportDirExists'], 'true'), eq(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.TargetBranch'], 'main'))
+                        continueOnError: false
+                        env:
+                          ADO_API_TOKEN: $(System.AccessToken)
+                          GITHUB_API_TOKEN: $(githubPublicRepoSecret)
+                          TARGET_BRANCH_NAME: '$(targetBranchName)'
+                          ADO_BUILD_ID: $(Build.BuildId)
+                        inputs:
+                          targetType: inline
+                          workingDirectory: '${{ parameters.buildDirectory }}'
+                          script: |
+                            Coverage_Report_Artifact_Name="Code Coverage Report"
+                            export CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_BASELINE=$Coverage_Report_Artifact_Name
+                            export CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_PR=$Coverage_Report_Artifact_Name
```

Comment:
We should refactor `flub report codeCoverage` and get rid of these variables. The command unconditionally adds the build id to the artifact name we pass here, so in reality it assumes that it's going to use the artifact produced by `PublishCodeCoverageResults` whose name we can't control, and that we're passing the correct prefix value here. "Code Coverage Report" is really the only value that would work, so having it be a variable doesn't make sense, IMO. And it'd be hard to make it take the full artifact name for both baseline and PR, because we'd have to find the correct builds here before invoking it. We can keep the flag in the flub command but with a default value and documentation saying "This is the only value that works so you don't need to override it, unless ADO eventually makes changes to the `PublishCodeCoverageResults` task".